reporter.go

```diff
@@ -0,0 +1,209 @@
+package k8sreporter
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"sync"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// NamespaceFilter is a filter function to choose what namespaces to dump.
+type NamespaceFilter func(string) bool
+
+// AddToScheme is a function for extend the reporter scheme and the CRs we are able to dump
+type AddToScheme func(*runtime.Scheme)
```

Comment:
I think this should return err as the respective AddToScheme we call does.

Reply:
done